components/builder-depot/src/server.rs

```diff
@@ -2512,6 +2532,7 @@ fn download_response_for_archive(archive: PackageArchive) -> IronResult<Response
     };
     response.headers.set(disp);
     response.headers.set(XFileName(archive.file_name()));
+    let _cleanup = remove_file(&archive.path);
```

Comment:
We shouldn't need this if we remove the pre-constructed paths

Reply:
Because of not using a tempfile or tempdir this is still obligatory. Re: my comment above

Reply:
👍